pkg/influx/auth_test.go

```diff
@@ -0,0 +1,119 @@
+package influx
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/influx2cortex/pkg/remotewrite"
+	"github.com/grafana/influx2cortex/pkg/remotewrite/remotewritemock"
+	"github.com/grafana/influx2cortex/pkg/server"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+)
+
+func TestAuthentication(t *testing.T) {
+	tests := []struct {
+		name            string
+		url             string
+		data            string
+		enableAuth      bool
+		orgID           string
+		expectedCode    int
+		remoteWriteMock func() *remotewritemock.Client
+	}{
+		{
+			name:         "test auth enabled valid org ID",
+			url:          "/write",
+			data:         "measurement,t1=v1 f1=2 1465839830100400200",
+			enableAuth:   true,
+			orgID:        "valid",
+			expectedCode: http.StatusNoContent,
+			remoteWriteMock: func() *remotewritemock.Client {
+				remoteWriteMock := &remotewritemock.Client{}
+				remoteWriteMock.On("Write", mock.Anything, mock.Anything).
+					Return(nil).Run(func(args mock.Arguments) {
+					ctx := args.Get(0).(context.Context)
+					orgID, err := user.ExtractOrgID(ctx)
+					require.NoError(t, err)
+					require.Equal(t, orgID, "valid")
+				})
+				return remoteWriteMock
+			},
+		},
+		{
+			name:         "test auth enabled invalid org ID",
+			url:          "/write",
+			data:         "measurement,t1=v1 f1=2 1465839830100400200",
+			enableAuth:   true,
+			orgID:        "",
+			expectedCode: http.StatusUnauthorized,
+			remoteWriteMock: func() *remotewritemock.Client {
+				remoteWriteMock := &remotewritemock.Client{}
+				remoteWriteMock.On("Write", mock.Anything, mock.Anything).Return(nil)
```

Comment:
Why is the mock Write function different here?  doesn't it still need to test the org ID to confirm that it's invalid?  I would have thought that the mocked Write function would be the same for each:
try to extract the orgID.  If none, return error
then validate the orgID.  If invalid, return error
else, return success

Then instead of putting the requirements in the mocked function itself, you're still testing the top-level return value as you had it before.  That way you're also testing that the code handles the error return value from the Write call correctly.